packages/manager/modules/emailpro/src/domain/dkim-autoconfig/emailpro-domain-dkim-autoconfig.controller.js

```diff
@@ -71,10 +70,53 @@ export default class EmailProDomainDkimAutoconfigCtrl extends DkimAutoConfigurat
   }
 
   hideConfirmButton() {
-    return (
-      this.dkimStatus === this.GLOBAL_DKIM_STATUS.NOT_CONFIGURED &&
-      this.domainDiag.isOvhDomain
-    );
+    const showWhen = [
+      this.GLOBAL_DKIM_STATUS.NOT_CONFIGURED,
+      this.GLOBAL_DKIM_STATUS.OK,
+      this.GLOBAL_DKIM_STATUS.DISABLED,
+    ];
+
+    return this.domainDiag.isOvhDomain && showWhen.includes(this.dkimStatus);
+  }
+
+  onFinishDkim() {
+    switch (this.dkimStatus) {
+      case this.GLOBAL_DKIM_STATUS.NOT_CONFIGURED:
+        return this.configDkim();
+      case this.GLOBAL_DKIM_STATUS.OK:
+        return this.deactivateDkim();
+      case this.GLOBAL_DKIM_STATUS.DISABLED:
+        return this.activateDkim();
+      default:
+        console.error('Invalid DKIM status:', this.dkimStatus);
+        return null;
+    }
+  }
+
+  activateDkim() {
+    const dkimSelectors = this.domain.dkim;
+
+    this.services.EmailProDomains.enableDkim(
+      this.services.$stateParams.productId,
+      this.domain.name,
+      dkimSelectors[0].selectorName,
+    ).finally(() => {
+      this.services.$scope.resetAction();
+    });
+  }
+
+  deactivateDkim() {
+    const dkimSelectors = this.domain.dkim;
+
+    this.services.EmailProDomains.disableDkim(
+      this.services.$stateParams.productId,
+      this.domain.name,
+      dkimSelectors[0].status === this.DKIM_STATUS.IN_PRODUCTION
+        ? dkimSelectors[0].selectorName
+        : dkimSelectors[1].selectorName,
+    ).finally(() => {
+      this.services.$scope.resetAction();
+    });
```

Comment:
This does seem to me quite the same function. Isn't it possible to refactor this and just have maybe pass a boolean somewhere to avoid repeat? I'll let you check, it's just a recommandation.

Reply:
Thanks for the suggestion on merging activateDkim and deactivateDkim. 
I believe keeping them separate offers clearer intent. 
I'd prefer readability over saving a few lines in this instance.